src/classes/AllocationParent.cls

```diff
@@ -0,0 +1,174 @@
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @group Allocations
+* @group-content ../../ApexDocContent/Allocations.htm
+* @description Holds behavior for Allocation__c parent objects.
+*/
+public inherited sharing class AllocationParent {
+    private Id id;
+    private SObjectType objectType = null;
+    private SObject record;
+    private Decimal donationAmount = 0;
+    private String currencyCode;
+    private String keyPrefix;
+    private String objectApiName;
+    private String objectLabelPlural;
+
+    private final String OPPORTUNITY_AMOUNT = String.valueOf(Opportunity.SObjectType.fields.Amount);
+    private final String PAYMENT_AMOUNT = String.valueOf(npe01__OppPayment__c.SObjectType.fields.npe01__Payment_Amount__c);
+    private final String OPPORTUNITY_LOOKUP = String.valueOf(Allocation__c.SObjectType.fields.Opportunity__c);
+    private final String CAMPAIGN_LOOKUP = String.valueOf(Allocation__c.SObjectType.fields.Campaign__c);
+    private final String RECURRING_DONATION_LOOKUP = String.valueOf(Allocation__c.SObjectType.fields.Recurring_Donation__c);
+    private final String PAYMENT_LOOKUP = String.valueOf(Allocation__c.SObjectType.fields.Payment__c);
+
+    public AllocationParent(Id parentId) {
+        id = parentId;
+        objectType = id.getSObjectType();
+        record = getRecord();
+        setPropertiesFromDescribe();
+
+        if (UserInfo.isMultiCurrencyOrganization()) {
+            currencyCode = (String)record.get(UTIL_Currency.CURRENCY_ISO_CODE_FIELD);
+        }
+    }
+
+    public Id getId() {
+        return id;
+    }
+
+    public SObjectType getSObjectType() {
+        return objectType;
+    }
+
+    public String getKeyPrefix() {
+        return keyPrefix;
+    }
+
+    public String getObjectLabelPlural() {
+        return objectLabelPlural;
+    }
+
+    public String getObjectApiName() {
+        return objectApiName;
+    }
+
+    public String getRecordName() {
+        return (String)record?.get('Name');
+    }
+
+    public String getCurrencyCode() {
+        return currencyCode;
+    }
+
+    public String getAllocationLookupApiName() {
+        return allocationLookupApiNameBySObjectType.get(objectType);
+    }
+
+    public Boolean isAnOpportunity() {
+        return objectType == Opportunity.SObjectType;
+    }
+
+    public Boolean isAPayment() {
+        return objectType == npe01__OppPayment__c.SObjectType;
+    }
+
+    public Boolean hasAnAmountField() {
+        return isAnOpportunity() || isAPayment();
+    }
+
+    public Boolean hasAnAmountGreaterThanZero() {
+        return hasAnAmountField() && getAmount() > 0;
+    }
+
+    public Decimal getAmount() {
```

Comment:
There are a few 'getters' in this class like this one that I had the same thoughts on as mentioned in the ALLO_ManageAllocations_CTRL class, essentially changing this query to amount() instead of getAmount(). Same for other methods with `getter` language here.

Reply:
Updated in the other locations, but this one is used by JS in the visualforce page. I don't think I want to get into refactoring the JS quite in this work, but open to hearing thoughts if the collective does!